packages/medusa/src/services/cart.ts

```diff
@@ -1424,12 +1427,13 @@ class CartService extends TransactionBaseService {
     })
 
     if (itemsToRemove.length) {
-      const results = await promiseAll(
-        itemsToRemove.map(async (item) => {
-          return this.removeLineItem(cart.id, item.id)
-        })
-      )
-      cart.items = results.pop()?.items ?? []
+      const itemIdsToRemove = itemsToRemove.map((item) => item.id)
+      await this.removeLineItem(cart.id, itemIdsToRemove)
+
+      for (const itemId of itemIdsToRemove) {
+        const index = cart.items.findIndex((lineItem) => lineItem.id === itemId)
+        cart.items.splice(index, 1)
+      }
```

Comment:
question: why do we splice here instead of filtering items? 
```suggestion
      const itemIdsToRemoveSet = new Set(itemIdsToRemove)

      cart.items = cart.items.filter((it) => !itemIdsToRemoveSet.has(it.id))
```